reporting-bugs/readme.md

```diff
@@ -0,0 +1,263 @@
+
+### Steps to Reproduce
+
+When you're writing a bug report, you need to be able to reproduce the bug. This means you need to be able to make it happen again. The person reading the report needs to be able to follow your steps and cause the same error.
+
+Here's an example of steps to reproduce:
+
+1. Go to my deployed site at [https://my-site.com](https://my-site.com)
+1. Click the "Login" button top right
+1. Choose "Sign in with GitHub"
+1. Authorise with GitHub and be redirected back to the site
+1. Login page error says `To confirm your identity, sign in with the same account you used originally.`
+
+### Bad bug reports
+
+<!--{{<note type="activity" title="Inversion, 5m" >}}-->
+
+1. Set a [timer for 5 minutes](https://www.google.com/search?q=timer+for+5+minutes).
+2. Popcorn around the room and each person give an example of a bad way to ask a question until the timer goes off. For example:
+
+> I was on the site, and then I clicked something, and it didn't work!
+
+<!--{{</note>}}-->
+
+### Why are these not useful?
+
+<!--{{<note type="activity" title="Inversion 2, 10m" >}}-->
+
+1.  Set a [timer for 10 minutes](https://www.google.com/search?q=timer+for+10+minutes).
+2.  Popcorn around the room and each person give a reason the previous question was bad until the timer goes off. For example:
+
+> What site are you on? What are you trying to do? How can I read your post and re-create the circumstances. Be specific! Share what you see.
+
+ <!--{{</note>}}-->
+
+### Expected behavior
+
+_What you expected to happen_
+
+When you're writing a bug report, you need to be able to explain what you expected to happen after you executed the steps you just described. You might feel this is so obvious that you don't need to write it down, but this assumption is drastically, drastically wrong.
+
+Here's an example of expected behavior:
+
+> I expected to be able to log in with my GitHub account and be redirected to the logged-in view of my dashboard on my site.
+
+### Actual behavior
+
+_What actually happened_
+
+Here's an example of actual behavior:
+
+> Instead, I was redirected back to the login page with an error message that says `To confirm your identity, sign in with the same account you used originally.`
+
+<!--{{<note type="activity" title="Spot the difference, 5m" >}}-->
+
+Look at the difference between the two examples above. What do you notice? _Apart from the user's reported problem_, what is the difference between the actual behaviour and the expected behaviour? It's subtle.
+
+1. Set a [timer for 3 minutes](https://www.google.com/search?q=timer+for+3+minutes).
+1. Popcorn around the room until you spot the difference.
+
+<!--{{</note>}}-->
+
+### Error messages
+
+_What the computer told you_
+
+Have you ever considered, and this is just a wild idea, reading the error messages that your computer gives when it goes wrong?
+
+Just kidding! We know all trainees are implacably opposed to reading error messages. But here's the thing:
```

Comment:
😭 U OK?

Reply:
I've long gone mad from having this conversation over and over! :wink: 

Actually what I'd love to do is build an error messages game/treasure hunt, where all the solutions are actually in the error messages, buried in crazy walls of text yelling OBJECT OBJECT [] ERR MIGR AB 191823733901:828749q01-xc, so trainees can practice finding the answer while not in a total panic.

Reply:
Quoting myself on a conversation I had recently about side benefits of TDD:

> I think there’s also a value to the fact that you’re constantly exposed to error messages that _aren’t (necessarily) unexpected_ - so many of the trainee devs I work with basically shut down when they get red text on the screen because it’s A Bad Thing, with TDD that becomes normal and hey ho guess I’ll read this one too.